### PR TITLE
[JFG-1116] Set Limits for the metrics in the step of scenario

### PR DESCRIPTION
**Pull request by a-badaev**

### CODE REVIEW

File: chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/limits/JLimitVsBaseline.java [hunk 1]
```diff
@@ -16,28 +17,53 @@ private JLimitVsBaseline(Builder builder) {
...
-    public static Builder builder(String metricName) {
-        return new Builder(metricName);
+    public static Builder builder(String metricId) {
+        return new Builder(metricId);
     }
-
+    
+    
+    /**
+     * Builder for {@link JLimit} to compare with current value.
```
Comment (dlatnikov) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/limits/JLimitVsBaseline.java]: to compare metrics, collected during {@link JHttpUserScenario} execution, with the baseline

File: chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/limits/JLimitVsBaseline.java [hunk 2]
```diff
@@ -16,28 +17,53 @@ private JLimitVsBaseline(Builder builder) {
...
      * @return builder for {@link JLimit}.
      */
-    public static Builder builder(JMetricName metricName) {
-        return new Builder(metricName.transformToString());
+    public static Builder builder(JMetricName metricId) {
+        return new Builder(metricId.transformToString());
+    }
+    
+    /**
+     * Builder for {@link JLimit} to compare with current value.
```
Comment (dlatnikov) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/limits/JLimitVsBaseline.java]: same as above

File: chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/limits/JLimitVsBaseline.java [hunk 3]
```diff
@@ -16,28 +17,53 @@ private JLimitVsBaseline(Builder builder) {
...
+     */
+    public static Builder builder(String scenarioId, String stepId, String metricId) {
+        return new Builder(StandardMetricsNamesUtil.generateScenarioStepMetricRegexp(scenarioId, stepId, metricId));
+    }
+    
     /**
      * Builder for {@link JLimit} to compare with baseline.
      *
-     * @param metricName name of standard metric.
+     * @param metricId name of standard metric.
```
Comment (dlatnikov) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/limits/JLimitVsBaseline.java]: unique id of the metric

File: chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/limits/JLimitVsRefValue.java [hunk 4]
```diff
@@ -21,45 +22,68 @@ private JLimitVsRefValue(Builder builder) {
     /**
      * Builder for {@link JLimit} to compare with current value.
```
Comment (dlatnikov) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/limits/JLimitVsRefValue.java]: with the reference value

File: chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/limits/JLimitVsRefValue.java [hunk 5]
```diff
@@ -21,45 +22,68 @@ private JLimitVsRefValue(Builder builder) {
...
      *
-     * @param metricName name of metric.
+     * @param metricId name of metric.
```
Comment (dlatnikov) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/limits/JLimitVsRefValue.java]: same as above

File: chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/limits/JLimitVsRefValue.java [hunk 6]
```diff
@@ -21,45 +22,68 @@ private JLimitVsRefValue(Builder builder) {
...
      * @return builder for {@link JLimit}.
      */
-    public static Builder builder(String metricName, RefValue refValue) {
-        return new Builder(metricName, refValue);
+    public static Builder builder(String metricId, RefValue refValue) {
+        return new Builder(metricId, refValue);
+    }
+    
+    /**
+     * Builder for {@link JLimit} to compare with current value.
```
Comment (dlatnikov) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/limits/JLimitVsRefValue.java]: same as above

here and everywhere

File: chassis/core/src/test/java/com/griddynamics/jagger/ExampleSimpleUserScenarioJLoadScenarioProvider.java [hunk 7]
```diff
@@ -46,12 +43,13 @@ public JLoadScenario exampleSimpleJaggerLoadScenarioUS() {
         JTerminationCriteria jTerminationCriteria = JTerminationCriteriaIterations.of(IterationsNumber.of(500), MaxDurationInSeconds.of(50));
 
         //TODO: JFG-1123
```
Comment (dlatnikov) [inline chassis/core/src/test/java/com/griddynamics/jagger/ExampleSimpleUserScenarioJLoadScenarioProvider.java]: please remove TODO

File: chassis/util/src/main/java/com/griddynamics/jagger/util/StandardMetricsNamesUtil.java [hunk 8]
```diff
@@ -210,4 +211,10 @@ public static String generateScenarioRegexp(String scenarioId) {
     public static String generateScenarioStepRegexp(String scenarioId, String stepId) {
         return String.format(SCENARIO_STEP_REGEXP_TEMPLATE, scenarioId, stepId);
     }
+    
+    public static String generateScenarioStepMetricRegexp(String scenarioId, String stepId, String metricId) {
+        return SCENARIO_STEP_METRIC_REGEXP_TEMPLATE.replace("{scenario_id}", Pattern.quote(scenarioId))
```
Comment (dlatnikov) [inline chassis/util/src/main/java/com/griddynamics/jagger/util/StandardMetricsNamesUtil.java]: why don't you use String.format like in other functions?

File: chassis/util/src/main/java/com/griddynamics/jagger/util/StandardMetricsNamesUtil.java [hunk 9]
```diff
@@ -158,6 +158,7 @@ public String getMetricId() {
     private static final String SCENARIO_STEP_REGEXP_TEMPLATE = "^.*" + USER_SCENARIO_ID + "%s" + US_STEP_ID + "\\d+_%s.*$";
     private static final String SCENARIO_REGEXP_TEMPLATE = "(^.*" + USER_SCENARIO_ID + "%s" + US_STEP_ID + ".*$)|(^.*%s.*(-sum|-Success rate|-Number of fails).*$)";
     private static final String DISPLAY_NAME_REGEXP = "^.*(" + ITERATIONS_SAMPLES + "|" + LATENCY_SEC + "|" + SUCCESS_RATE + ").*";
+    private static final String SCENARIO_STEP_METRIC_REGEXP_TEMPLATE = "^.*" + USER_SCENARIO_ID + "{scenario_id}" + US_STEP_ID + "\\d+_{step_id}_" + US_METRIC_ID + "{metric_id}_(-.*)?$";
```
Comment (dlatnikov) [inline chassis/util/src/main/java/com/griddynamics/jagger/util/StandardMetricsNamesUtil.java]: are you sure about this regexp. looks strange.

Comment (a-badaev) [inline chassis/util/src/main/java/com/griddynamics/jagger/util/StandardMetricsNamesUtil.java]: Pretty confident

File: chassis/util/src/main/java/com/griddynamics/jagger/util/StandardMetricsNamesUtil.java (hunk 8 above)
Comment (a-badaev) [inline chassis/util/src/main/java/com/griddynamics/jagger/util/StandardMetricsNamesUtil.java]: it is more reliable and human readable

**Pull request merged**